### PR TITLE
fix: push notification cancelation

### PR DESCRIPTION
**Pull request by Brazol**

### CODE REVIEW

File: packages/stream_video/example/lib/main.dart [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import 'package:stream_video/stream_video.dart';
+
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+
+  StreamVideo.create(
+    'api_token',
+    user: User.guest(userId: 'user_id'),
+  );
+}
```
Comment (deven98) [inline packages/stream_video/example/lib/main.dart:11]: Is this an example for pub?

File: packages/stream_video/lib/src/stream_video.dart [hunk 2]
```diff
@@ -572,13 +572,18 @@ class StreamVideo {
     final callCid = payload['call_cid'] as String?;
     if (callCid == null) return false;
 
-    final uuid = const Uuid().v4();
+    var callId = const Uuid().v4();
+    final splitCid = callCid.split(':');
+    if (splitCid.length == 2) {
+      callId = splitCid.last;
+    }
+
```
Comment (deven98) [inline packages/stream_video/lib/src/stream_video.dart:580]: Are we sure it's in the correct format? I remember this giving me some troubles earlier.

File: packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart [hunk 3]
```diff
@@ -54,14 +55,26 @@ class StreamVideoPushNotificationManager implements PushNotificationManager {
...
+            ),
+          );
+
+          _subscriptions.add(
+            _idCallAccepted,
+            client.events.on<CoordinatorCallAcceptedEvent>(
+              (event) async {
+                await FlutterCallkitIncoming.silenceEvents();
+                await FlutterCallkitIncoming.endCall(event.callCid.id);
+                await Future<void>.delayed(const Duration(milliseconds: 300));
```
Comment (deven98) [inline packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart:76]: Why do we need to delay here?

File: packages/stream_video/example/lib/main.dart (hunk 1 above)
Comment (Brazol) [reply packages/stream_video/example/lib/main.dart:11]: yes, we had it before just like this (probably just for the pub scoring) then it got removed and scoring with it. So now it's back :)

File: packages/stream_video/lib/src/stream_video.dart (hunk 2 above)
Comment (Brazol) [reply packages/stream_video/lib/src/stream_video.dart:580]: yes, the issue is with iOS CallKit requiring it to be valid uuid, which it is when we split it and use the id part and not the whole cid

File: packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart (hunk 3 above)
Comment (Brazol) [reply packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart:76]: To account for asynchronous race, it happened that events were unsilenced before the endCall event was triggered on the native side.

File: packages/stream_video/example/lib/main.dart (hunk 1 above)
Comment (deven98) [reply packages/stream_video/example/lib/main.dart:11]: This is fine for now but let's add these things in separate PRs so we don't confuse stuff.

File: packages/stream_video/lib/src/stream_video.dart (hunk 2 above)
Comment (deven98) [reply packages/stream_video/lib/src/stream_video.dart:580]: For some reason, in the early video time it sometimes wasn't a valid UUID, good to know it is now. This is great, thanks.

File: packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart (hunk 3 above)
Comment (deven98) [reply packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart:76]: Fair enough.

**Review by deven98 — APPROVED** [review]
LGTM

**Pull request merged**